### PR TITLE
Parse macro invocations containing $

### LINKED ISSUE #95: Parse error on $ in macro invocations
Example:
```rust
use syn::Token;
use quote::ToTokens;

fn main() {
    let dollar_sign: Token![$] = Default::default();
    println!("{}", dollar_sign.into_token_stream());
}
```
This is [valid, runnable rust code](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=9b6630ddab123383a6b714c843f1608d), but does not parse.

This occurs because the `macro_invocation` grammar production ultimately delegates to `_non_special_token`, which excludes `$`. This is the desired bevarior when parsing `macro_rules!` declarations, where `$` is assigned special meaning, but not when parsing other macro invocations, where `$` is just a normal token.

This currently breaks Atom rust syntax highlighting for the scope containing the macro invocation and the rest of the document thereafter.

### PR DESCRIPTION
**Pull request by ninevra**
Closes #95.

Adds a set of productions for general delimited token trees. These productions do not attempt to find `macro_rules!`-specific features like repetitions, and consider `$` to be a normal token. For backwards compatibility, use of the new productions aliases them to `token_tree`.

Blocks draft #93, which adds similar productions; #93 could be rebased on top of this if relevant.

### CODE REVIEW

File: grammar.js [hunk 1]
```diff
@@ -879,7 +879,23 @@ module.exports = grammar({
...
+    ),
+
+    _delim_tokens: $ => choice(
+      $._non_delim_token,
+      alias($.delim_token_tree, $.token_tree),
+    ),
+
+    _non_delim_token: $ => choice(
+      $._non_special_token,
+      '$'
```
Comment (aryx) [inline grammar.js:898]: do we need other characters here? This is needed just for '$'?

**Review by aryx — CHANGES_REQUESTED** [review]
Could you rebase on the latest so there is no conflict?

**Comment by ninevra** [comment]
Sorry, I've been away from github for a few months. I'll take a look at this and see if I can rebase & address comments soon.

File: grammar.js (hunk 1 above)
Comment (ninevra) [reply grammar.js:916]: I believe `$` is the only character we need to add here. `_non_special_token` is analogous to the first branch of the [`MacroMatch` construction](https://doc.rust-lang.org/reference/macros-by-example.html#macros-by-example) in the reference, which allows any token other than `$` and delimiters.

**Comment by ninevra** [comment]
Rebased to resolve merge conflicts.

Removed special handling of metavariables (`$ident`, `$crate`) in macro invocations, as those are specific to macro transcribers.

**Review by aryx — APPROVED** [review]

**Comment by aryx** [comment]
This looks good! I guess this can unblock your other PRs now!

**Comment by ninevra** [comment]
Thanks for merging! I'll work on rebasing #93.

**Pull request merged**